app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java

```diff
@@ -1254,20 +1256,20 @@ public void onResume() {
           if (Build.VERSION.SDK_INT >= VERSION_CODES.M) {
             i.putExtra(Intent.EXTRA_PROCESS_TEXT, intent.getStringExtra(Intent.EXTRA_PROCESS_TEXT));
           }
-          intent.setAction("");
+          intent.setAction(null);
           startActivityForResult(i, REQUEST_FILE_SEARCH);
           break;
         }
         case KiwixSearchWidget.TEXT_CLICKED:
-          intent.setAction("");
+          intent.setAction(null);
           goToSearch(false);
           break;
         case KiwixSearchWidget.STAR_CLICKED:
-          intent.setAction("");
+          intent.setAction(null);
           goToBookmarks();
           break;
         case KiwixSearchWidget.MIC_CLICKED:
-          intent.setAction("");
+          intent.setAction(null);
```

Comment:
If the action is just being set to null is there even a point of setting it at all?

Reply:
As far as I know, An empty string is a string instance of zero length, whereas a null string has no value at all. So when we opening widget second time after opening activity in the background it doesn't come to main search screen.

Reply:
After looking at the code I saw that the source of the intent was `Activity.getIntent()`, weird that we are clearing the intent's action and we are in fact not even saving this as there is no corresponding call to `Activity.setIntent`.
Please delete all the setAction calls. They are most likely an error

Reply:
Ah I think I know what it is now, this code is "duplicated" in onResume and in onNewIntent.
It needs to be in onCreate and onNewIntent, both should have a method that handles the intent and depending on the action launches the correct activity

Reply:
To make the matter more confusing onCreate already has intent handling but not based on the action, instead by extras like onNewIntent.
Even more confusingly I cannot see where these extras get set.